src/main/java/com/example/SS2_Backend/util/SampleDataGenerator.java

```diff
@@ -62,119 +85,175 @@ public static void main(String[] args) {
-//                    problem.getIndividuals().getCapacities());
-//            System.out.println("Solution has duplicate individual? : " + tester.hasDuplicate());
         }
         System.out.println("\nExecution time: " + runtime + " Second(s) with Algorithm: " + algo);
 
     }
+
     /**
-     * Constructor for configuring generator with required sets.
+     * Generates a NewStableMatchingProblemDTO instance based on the configured parameters.
      *
-     * @param numberOfSet1 Number of individuals in set 1
-     * @param numberOfSet2 Number of individuals in set 2
+     * @return A NewStableMatchingProblemDTO object
      */
-    public SampleDataGenerator(int numberOfSet1, int numberOfSet2) {
-        this.numberOfSet1 = numberOfSet1;
-        this.numberOfSet2 = numberOfSet2;
+    public NewStableMatchingProblemDTO generateDto() {
+        NewStableMatchingProblemDTO problemDTO = new NewStableMatchingProblemDTO();
+        problemDTO.setNumberOfIndividuals(individualNum);
+        problemDTO.setNumberOfSets(numberForeachSet.length);
+        problemDTO.setNumberOfProperty(numberOfProperties);
+        problemDTO.setIndividualSetIndices(generateSetIndices());
+        problemDTO.setIndividualCapacities(generateCapacities());
+        problemDTO.setIndividualProperties((double[][]) generatePW().get("property"));
+        problemDTO.setIndividualWeights((double[][]) generatePW().get("weight"));
+        problemDTO.setIndividualRequirements(generateRequirementString());
+        problemDTO.setEvaluateFunctions(evaluateFunctions);
+        problemDTO.setFitnessFunction(fnf);
+        return problemDTO;
     }
 
     /**
-     * Generates a StableMatchingProblem instance based on the configured parameters.
+     * Generates a StableMatchingRBOProblem instance based on the configured parameters.
      *
-     * @return A StableMatchingProblem object
+     * @return StableMatchingRBOProblem
      */
-    public StableMatchingProblem generate() {
-        // Generate the individual population
-        ArrayList<Individual> individuals = generateIndividualsWithCapacity();
-
-//        // Define excluded pairs
-//        int[][] excludedPairs = {
-//                {0, 3},
-//                {1, 2},
-//                {0, 2},
-//                {1, 3}
-//        };
-        // Create and configure the StableMatchingProblem instance
-        StableMatchingProblem problem = new StableMatchingProblem();
-        problem.setEvaluateFunctionForSet1(f1);
-        problem.setEvaluateFunctionForSet2(f2);
-        problem.setFitnessFunction(fnf);
-        problem.setPopulation(individuals, this.propNames, null);
-        return problem;
+    public MatchingProblem generateProblem() {
+        MatchingProblem matchingProblem;
+
+        NewStableMatchingProblemDTO newDto = this.generateDto();
+        switch (this.matchingProblemType) {
+            case MTM -> {
+                this.capRandomize = new boolean[]{true, true};
```

Comment:
bỏ this.cap ... ở mỗi case trong switch đi nhé, trong hàm này đây là bước generate rồi, không còn là bước config nữa

Reply:
Thế phần này set như nào thế anh? Tức là mình phải viết trong Constructor cho tự generate capRandomize cho từng cases (MTM, OTM, OTO) ạ?